File: Desarrollo_de_apps/Practica2/JAVA/ProyectoBarcos/src/proyectobarcos/Barco.java
package proyectobarcos;
//Clase principal que implementa la interface Navegación
public class Barco implements Navegacion{
    //Atributos de la clase
    String nombre;
    int noVelas;
    boolean ancla;
    //Constructor primario (sin argumentos)
    public Barco(){
        nombre = "Sin nombre";
        noVelas = 2;
    }
    //Constructor sobrecargado (con 2 argumentos)
    public Barco(String nombre, int noVelas){
        //this();
        this.nombre = nombre;
        this.noVelas = noVelas;
    }
    //Métodos sobrescritos y propios de la interface
    @Override
    public void avanzar() {
        System.out.println("El " + nombre + " está avanzando");
    }

    @Override
    public void girarEstribor() {
        System.out.println("El " + nombre + " gira a estribor");
    }

    @Override
    public void girarBabor() {
        System.out.println("El " + nombre + " gira a babor");
    }

    @Override
    public boolean anclar() {
        ancla = false;
        return ancla;
    }

    @Override
    public boolean desanclar() {
        ancla = true;
        return ancla;
    }
    //Sobreescritura del método ToString
    @Override
    public String toString() {
        return "Nombre: " + nombre + " | " + "No. de velas: " + noVelas;
    }
}

File: Desarrollo_de_apps/Practica3/ProyectoBarco/ProyectoBarco/Barco.cs
using System;
namespace ProyectoBarco
{
    //Clase principal que implementa la interface Navegación
    public class Barco : INavegacion
    {
        //Atributos de la clase
        public String nombre;
        public int noVelas;
        public bool ancla;

        //Constructor primario (sin argumentos)
        public Barco()
        {
            nombre = "Sin nombre";
            noVelas = 2;
        }

        //Constructor sobrecargado (con 2 argumentos)
        public Barco(String nombre, int noVelas)
        {
            this.nombre = nombre;
            this.noVelas = noVelas;
        }

        //Métodos propios de la interface
        public bool Anclar()
        {
            ancla = false;
            return ancla;
        }

        public void Avanzar()
        {
            Console.WriteLine("El " + nombre + " está avanzando");
        }

        public bool Desanclar()
        {
            ancla = true;
            return ancla;
        }

        public void GirarBabor()
        {
            Console.WriteLine("El " + nombre + " gira a babor");
        }

        public void GirarEstribor()
        {
            Console.WriteLine("El " + nombre + " gira a estribor");
        }

        //Sobreescritura del método ToString
        public override String ToString()
        {
            return "Nombre: " + nombre + " | " + "No. velas: " + noVelas;
        }
    }
}
